drivers/adc/ad7606/ad7606.h

```diff
@@ -318,12 +327,18 @@ struct ad7606_dev {
 };
 
 /**
- * @struct ad7606_dev
+ * @struct ad7606_init_param
  * @brief Device driver initialization parameters
  */
 struct ad7606_init_param {
 	/** SPI initialization parameters */
 	struct no_os_spi_init_param spi_init;
+#ifdef XILINX_PLATFORM
```

Comment:
i think we should not pollute the device driver code with #ifdefs that distinguish hardware platforms, device driver shouldn't care much about what hardware it runs on

in this case it just needs to know that it should or should not use an axi_clkgen (which could be intel, could be xilinx etc.). i suggest to remove all the ifdefs and operate on the fact that clkgen_init is populated or not, this way you move the #ifdef logic to the app level code, which indeed should care about what exact hw platform we're running on

Reply:
we can do it like that as well;
this was copied from `ad7616.c`, which doesn't mean that we should use that model;

i'll probably need to re-consider how to group them;
the only benefit these `#ifdef` guards have, is that there is a clear separation between "simple" mode for this driver and FPGA-mode;
the initial version of this driver, seems to have been developed as directly connected to an SoC;